### ISSUE TITLE
Fog-Mime / Fog-Json

### ISSUE THREAD

**Issue by karlfreeman** [fog/fog#2732]
TLDR; would a pull request creating `Fog-Mime` like [`Fog-Json`](https://github.com/fog/fog-json) be something worthwhile?

**Is `mime-types` only in `Fog` for `Fog::Local`? Can't seem to see where else its used?**

~~Whilst putting together [`MultiSync`](https://github.com/karlfreeman/multi_sync) I became frustrated with the slight variances in MIME type detection which led to me to write [`MultiMime`](https://github.com/karlfreeman/multi_mime) (as you can imagine its heavily inspired by [`MultiJson`](https://github.com/intridea/multi_json)).~~

~~It appears digging into Fog a bit more that it internally sets the [`content_type`](https://github.com/fog/fog/blob/e1770354c9be10cf863c4f9371d851efc077a462/lib/fog/aws/models/storage/file.rb#L217) on S3 PUT via the [file's `content_type` method](https://github.com/fog/fog/blob/1f12a196e37d0038a5116b7c609e65882a741ee6/lib/fog/local/models/storage/file.rb#L30-L36).~~

~~If the decision here is that [MIME::Types](https://github.com/halostatue/mime-types) is the best MIME type detection then I could only agree but just like Json there are edge cases where it would be good to offer some flexabiity? Would Fog consider MIME type detection the same as Json and extract it from Fog's core?~~.

Not applicable as this appears to just be for `Fog::Local` :unamused:.
